dissect/target/target.py

```diff
@@ -285,10 +282,14 @@ def _find(find_path: Path):
                         # Attempt to load the target using this loader
                         target = cls._load(sub_entry, ldr)
                         loaded = True
+                        at_least_one_loaded = True
                         yield target
-                    except Exception as e:
-                        getlogger(sub_entry).error("Failed to load target with loader %s", ldr, exc_info=e)
-                        continue
+
+                        # If DirLoader, ignore further entries. Otherwise search children/nested
+                        if isinstance(ldr, DirLoader):
```

Comment:
Should it not be a continue then, you only want to skip children right? One indent to the left only moves it out of the try btw.

Reply:
No, "children" here are nested targets, such as nested virtual machine files. That is very much still a valid use case for directory targets.

Reply:
I do not understand this, the Dirloader expects a directory reflecting a target, it should not open children because this interferes with the Dirloader concept itself. For instance, a subdir /system may either contain a new Dirloadable target or be part of the current Dirtarget. So I don't think this is a good idea. Maybe I should make a test implementation so we can test this scenario, I always have trouble implementing hypothetical functionality, hard to wrap my head around, so I'll craft a little example to gain better understanding.

Reply:
Those are not the same thing - "children" in this context are those detected by the `ChildPlugin`'s, e.g. such as this one: https://github.com/fox-it/dissect.target/blob/main/dissect/target/plugins/child/hyperv.py. Not "child directories".

A directory target could contain these `.vmcx` and `.vhdx` files, so child targets are still relevant to directory targets.

Reply:
O, I see. I knew they were VMs but I did not know it was really a hard limitation, I thought people could write their own child plugins as well and they could be anything conceptually. I would consider renaming this to VMs (and vm-plugins) then, because child may lead to a more open interpretation (as we have just witnessed - even while I knew the "current" implementations were only VMs).

Given these facts, I suggest we remove the dirloader check completely. Indenting it to the left will break stuff because it will also break on a failed ldr (no check for loaded flag). And... if we fix that we end up with duplicate code with the line below that already breaks if the top-level is loaded.

For completeness and understanding I added a 5th scenario to demonstrate that the hypothetical Dir-based loader will work as expected given the current update. Feel free to request any additional scenarios or test cases. If I overlook something else feel free to point it out to me. Sorry for not the misunderstanding on my side.

Reply:
VM's are just an example though, they don't necessarily have to be VM's. For example, they can also be Docker containers or [Virtuozzo](https://github.com/fox-it/dissect.target/blob/main/dissect/target/plugins/child/virtuozzo.py) containers. People can indeed write their own child plugins and it can conceptually be anything, but it's anything _within_ a target. If a `DirLoader` results in a valid target with a valid Windows or Linux root filesystem, anything that can be in there that can be seen as a nested target (VM, container, whatever) can be loaded by a child plugin.

Reply:
Ok fair enough.